src/DeepLinkModule/javasource/deeplink/actions/ExecuteDeeplink.java

```diff
@@ -115,8 +115,9 @@ public java.lang.Boolean executeAction() throws Exception
 
 			//remove the pendinglink, unless it should be reused during this session..
 			if (link.getUseAsHome()) { //do not remove if used as home.
-				this.pendinglink.setSessionId(this.getContext().getSession().getId().toString());
-				Core.commit(this.getContext(), this.pendinglink.getMendixObject());
+				IContext sudoContext = getContext().createSudoClone();
+				this.pendinglink.setSessionId(sudoContext, this.getContext().getSession().getId().toString());
+				Core.commit(sudoContext, this.pendinglink.getMendixObject());
 			}
 			else {
 				Core.delete(this.getContext(), this.pendinglink.getMendixObject());

```

Comment:
All user roles still have object deletion rights. This could be changed which would mean sudoContext would need to be used here, but we discussed in the refinement there was no need for that.